src/app/api/tasks/tasks.service.ts

```diff
@@ -578,12 +578,16 @@ export class TasksService extends BaseService {
 
     // Case 4
     // --- Handle task moved from completed to incomplete IU logic
+    const isAssignedIU =
```

Comment:
since we are checking if the assignee is currentUser, `isSelfAssignedIU` makes sense?